src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -231,6 +231,175 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNew
     return *newProfile;
 }
 
+// Method Description:
+// - Duplicate a new profile based off another profile's settings
+// Arguments:
+// - source: the Profile object we are duplicating (must not be null)
+// Return Value:
+// - a reference to the new profile
+winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::DuplicateProfile(Model::Profile source)
+{
+    THROW_IF_NULL_ALLOC(source);

```

Comment:
this isn't quite the right error code -- it maps more closely to "we ran out of memory, because an allocation failed"

Reply:
Fixed!